ast/json_test.go

```diff
@@ -157,6 +157,308 @@ func TestJSONMarshal(t *testing.T) {
 			},
 			want: `{"type":"BuiltinStatement","id":{"type":"Identifier","name":"task"}}`,
 		},
+		{
+			name: "NamedType",
+			node: &ast.NamedType{
+				BaseNode: ast.BaseNode{},
+				ID: &ast.Identifier{
+					BaseNode: ast.BaseNode{},
+					Name:     "int",
+				},
+			},
+			want: `{"type":"NamedType","name":{"type":"Identifier","name":"int"}}`,
+		},
+		{
+			name: "TvarType",
+			node: &ast.TvarType{
+				BaseNode: ast.BaseNode{},
+				ID: &ast.Identifier{
+					BaseNode: ast.BaseNode{},
+					Name:     "A",
+				},
+			},
+			want: `{"type":"TvarType","name":{"type":"Identifier","name":"A"}}`,
+		},
+		{
+			name: "ArrayType",
+			node: &ast.ArrayType{
+				BaseNode: ast.BaseNode{},
+				ElementType: &ast.ArrayType{
+					BaseNode: ast.BaseNode{},
+					ElementType: &ast.NamedType{
+						BaseNode: ast.BaseNode{},
+						ID: &ast.Identifier{
+							BaseNode: ast.BaseNode{},
+							Name:     "int",
+						},
+					},
+				},
+			},
+			// r#"{"type":"ArrayType","element":{"type":"ArrayType","element":{"type":"NamedType","name":{"type":"Identifier","name":"A"}}}}"#
+			want: `{"type":"ArrayType","elementType":{"type":"ArrayType","elementType":{"type":"NamedType","name":{"type":"Identifier","name":"int"}}}}`,
+		},
+		{
+			name: "RecordType",
+			node: &ast.RecordType{
+				BaseNode: ast.BaseNode{},
+				Properties: []*ast.PropertyType{
+					{
+						BaseNode: ast.BaseNode{},
+						Name: &ast.Identifier{
+							BaseNode: ast.BaseNode{},
+							Name:     "A",
+						},
+						Ty: &ast.NamedType{
+							BaseNode: ast.BaseNode{},
+							ID: &ast.Identifier{
+								BaseNode: ast.BaseNode{},
+								Name:     "int",
+							},
+						},
+					},
+				},
+				Tvar: &ast.Identifier{
+					BaseNode: ast.BaseNode{},
+					Name:     "A",
+				},
+			},
+			want: `{"type":"RecordType","properties":[{"type":"PropertyType","name":{"type":"Identifier","name":"A"},"ty":{"type":"NamedType","name":{"type":"Identifier","name":"int"}}}],"tvar":{"type":"Identifier","name":"A"}}`,
+		},
+		{
+			name: "RecordType_NoTvar_NoProp",
+			node: &ast.RecordType{
+				BaseNode:   ast.BaseNode{},
+				Properties: []*ast.PropertyType{},
+				Tvar:       nil,
+			},
+			//{"type":"RecordType","properties":[]}"#);
+			want: `{"type":"RecordType","properties":[],"tvar":null}`,
```

Comment:
Why does `tvar` get added to the json string?

Reply:
I get an error saying I need to add something for tvar there, unless there is something I need to change to make it optional?

Reply:
My only concern is if we can unmarshal this string to the appropriate Rust type. As long as we can do that and Rust isn't confused by `"tvar":null` that's all that matters.